embeddings/hyperparameter_search/configspace.py

```diff
@@ -0,0 +1,191 @@
+    )
+    max_epochs: Parameter = SearchableParameter(
+        name="max_epochs", type="int_uniform", low=1, high=5, step=1
+    )
+
+    @staticmethod
+    def _parse_model_trainer_parameters(
+        parameters: Dict[str, PrimitiveTypes]
+    ) -> Tuple[Dict[str, PrimitiveTypes], Dict[str, PrimitiveTypes]]:
+        task_train_keys: Final = {
+            "learning_rate",
+            "mini_batch_size",
+            "max_epochs",
+            "param_selection_mode",
+            "save_final_model",
+        }
+        task_train_kwargs = {k: parameters.pop(k) for k in task_train_keys if k in parameters}
+        return parameters, task_train_kwargs
+
+
+class FlairModelTrainerConfigSpace(AbstractFlairModelTrainerConfigSpace):
+    @staticmethod
+    def parse_parameters(parameters: Dict[str, PrimitiveTypes]) -> SampledParameters:
+        (
+            parameters,
+            task_train_kwargs,
+        ) = FlairModelTrainerConfigSpace._parse_model_trainer_parameters(parameters)
+        if len(parameters):
+            raise ValueError(
+                f"Some of the parameters are not mapped. Unmapped parameters: {parameters}"
+            )
+        return {"task_train_kwargs": task_train_kwargs}
+
+
+@dataclass
+class SequenceLabelingConfigSpace(AbstractFlairModelTrainerConfigSpace):
+    hidden_size: Parameter = SearchableParameter(
+        name="hidden_size", type="int_uniform", low=128, high=2048, step=128
+    )
+    use_rnn: Parameter = SearchableParameter(
+        name="use_rnn", type="categorical", choices=[True, False]
+    )
+    rnn_type: Parameter = SearchableParameter(
+        name="rnn_type", type="categorical", choices=["LSTM", "GRU"]
+    )
+    rnn_layers: Parameter = SearchableParameter(
+        name="rnn_layers", type="int_uniform", low=1, high=3, step=1
+    )
+    dropout: Parameter = SearchableParameter(
+        name="dropout", type="discrete_uniform", low=0.0, high=0.5, q=0.05
+    )
+    word_dropout: Parameter = SearchableParameter(
+        name="word_dropout", type="discrete_uniform", low=0.0, high=0.5, q=0.05
+    )
+    locked_dropout: Parameter = SearchableParameter(
+        name="locked_dropout", type="discrete_uniform", low=0.0, high=0.5, q=0.05
+    )
+    reproject_embeddings: Parameter = SearchableParameter(
+        name="reproject_embeddings", type="categorical", choices=[True, False]
+    )
+    use_crf: Parameter = SearchableParameter(
+        name="use_crf", type="categorical", choices=[True, False]
+    )
+
+    def sample_parameters(self, trial: optuna.trial.Trial) -> Dict[str, PrimitiveTypes]:
+        parameters = {}
+
+        use_rnn_name, use_rnn_val = self._parse_parameter(trial=trial, param_name="use_rnn")
+        parameters[use_rnn_name] = use_rnn_val
+
+        if use_rnn_val:
+            for rnn_param in ("rnn_layers", "rnn_type"):
+                parameters.update([self._parse_parameter(trial=trial, param_name=rnn_param)])
```

Comment:
what do you think about separating this out into a distinct method like _map_task_specific_parameters? it'd be IMO more transparent